hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/PipelinePlacementPolicy.java

```diff
@@ -99,9 +99,11 @@ boolean meetCriteria(DatanodeDetails datanodeDetails, int nodesRequired) {
       try {
         pipeline = stateManager.getPipeline(pid);
       } catch (PipelineNotFoundException e) {
-        LOG.error("Pipeline not found in pipeline state manager during" +
-            " pipeline creation. PipelineID: " + pid +
-            " exception: " + e.getMessage());
+        if (LOG.isDebugEnabled()) {
```

Comment:
I don't think we need to wrap the debug logs in `if (LOG.isDebugEnabled())` if we use the SLF4J style logs with {} placeholders rather than string interpolation. The log message will only be evaluated if the debug level is set.

For this log, I would suggest using the {} placeholders and let the full stack trace be logged rather than just `e.getMessage()`, eg:

```
LOG.debug("Pipeline not found in pipeline state manager during" +
              " pipeline creation. PipelineID: {}", pid, e);
```

Reply:
Thanks for the info. How about the logs without "{}" ? Shall we check if debug is enabled?

Reply:
If you don't use the {} syntax, then it makes sense to wrap the log in the if statement to avoid needing to construct the string. Ideally you should use the {} syntax for the logs, as they are easier to write, generally easier to read and more efficient.

Reply:
I added {} and removed isDebugEnabled check

Reply:
The other logs have been changed, but this one is still using interpolation and is wrapped in the if statement. Could you change it too please?

With SLF4J, if you pass an exception object as the last parameter it will print the full stack into the log automatically, eg:
```
LOG.debug("Pipeline not found in pipeline state manager during" +
              " pipeline creation. PipelineID: {}", pid, e);
```
I don't think the full stack will get printed using string concatenation as that will call toString on the exception which will just return the message part and not the stack trace.

Reply:
Yea just updated. Thanks.